### PR TITLE
Do not run query if dashboard filter is not connected to the card

### LINKED ISSUE #32444: Dashboard cards which contain a filter at question level gets refreshed even if Dashboard Filter is not linked to the card
### Describe the bug

Seems to be related to https://github.com/metabase/metabase/issues/8030. 

In this case though cards are getting refreshed when a filter that is not linked to them is changed. Note this happens only for cards that are filtered at question level

Important that the behavior in-app dashboard is also reflected on Public and Embedding, so it doesn't divert in behavior.

### To Reproduce

1. Go to Question -> Sample DB -> Orders -> Filter by Total is not Empty

image

2. Add it to a Dashboard
3. Add any type of Filter but don't link it to anything

image

4. Change Value of the filter and see question getting refreshed.

NOTE: This doesn't happen for cards that don't have a filter at question level

### Expected behavior

Question is not refreshed if no filter is linked to it

### Logs

None

### Information about your Metabase installation

```JSON
1.46.6 and master
```

### Severity

Would say major since it's a performance issue and a regression (or a partial fix https://github.com/metabase/metabase/issues/8030)) 

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by uladzimirdev**
Closes https://github.com/metabase/metabase/issues/32444

### Description

it's a MBQLv1 bug of query normalization. In this PR we replaced the old (wrong) logic of building filter with new (correct).

### How to verify

Describe the steps to verify that the changes are working as expected.

1. Go to Question -> Sample DB -> Orders -> Filter by Total is not Empty...
2. Add it to a Dashboard
3. Add any type of Filter but don't link it to anything

### Demo

https://www.loom.com/share/d88157488c6745ff8ecb23861ff17991

### Checklist

- [x] Tests have been added/updated to cover changes in this PR

### CODE REVIEW

File: frontend/src/metabase-lib/queries/utils/index.js [hunk 1]
```diff
@@ -70,7 +70,7 @@ export function cleanQuery(query) {
     _.all(filter, a => a != null),
   );
   if (filters.length > 0) {
-    query.filter = ["and", ...filters];
+    query.filter = QUERY.getFilterClause(filters);
```
Comment (kamilmielnik) [inline frontend/src/metabase-lib/queries/utils/index.js:73]: nit: if this was TypeScript, we'd get an error here: `QUERY.getFilterClause` can return `undefined`, but `query.filter` must be an array (or be absent from the object).
This case is impossible due to `if (filters.length > 0)` though.

**Review by kamilmielnik — APPROVED** [review]
Looks and works well :+1:

File: frontend/src/metabase-lib/queries/utils/index.js (hunk 1 above)
Comment (uladzimirdev) [reply frontend/src/metabase-lib/queries/utils/index.js:73]: good catch, I met such situations several times, but it's going to be a problem of someone in the future when he decided to switch those files to TS 😅

File: e2e/test/scenarios/dashboard-filters/reproductions/8030-32444-reload-card-without-change.cy.spec.js [hunk 2]
```diff
@@ -57,6 +76,38 @@ describe("issue 8030", () => {
   });
 });
 
+describe("issue 32444", () => {
+  beforeEach(() => {
+    restore();
+    cy.signInAsAdmin();
+  });
+
+  it("should not reload dashboard cards with filter not connected to a filter (metabase#32444)", () => {
```
Comment (kulyk) [inline e2e/test/scenarios/dashboard-filters/reproductions/8030-32444-reload-card-without-change.cy.spec.js:85]: > with filter not connected to a filter

Probably a typo? Should it be something like "with a dashcard not connected to a filter" or vice-versa?

File: e2e/test/scenarios/dashboard-filters/reproductions/8030-32444-reload-card-without-change.cy.spec.js [hunk 3]
```diff
@@ -120,3 +171,9 @@ const interceptRequests = ({ dashboard_id, card1_id, card2_id }) => {
     `/api/dashboard/${dashboard_id}/dashcard/*/card/${card2_id}/query`,
   ).as("getCardQuery2");
 };
+
+function addFilterValue() {
+  filterWidget().click();
+  cy.findByPlaceholderText("Enter some text").type("5{enter}");
+  cy.button("Add filter").click();
+}
```
Comment (kulyk) [inline e2e/test/scenarios/dashboard-filters/reproductions/8030-32444-reload-card-without-change.cy.spec.js:179]: WDYT about adding a `value` argument to `addFilterValue`? Seeing it called without any arguments made me think we're not actually filling any value

**Review by kulyk — APPROVED** [review]
LGTM, just two minor comments around the E2E test

**Review by ranquild — APPROVED** [review]

File: e2e/test/scenarios/dashboard-filters/reproductions/8030-32444-reload-card-without-change.cy.spec.js (hunk 3 above)
Comment (uladzimirdev) [reply e2e/test/scenarios/dashboard-filters/reproductions/8030-32444-reload-card-without-change.cy.spec.js]: I had the similar idea, but wasn't sure about it. fixed!

File: e2e/test/scenarios/dashboard-filters/reproductions/8030-32444-reload-card-without-change.cy.spec.js (hunk 2 above)
Comment (uladzimirdev) [reply e2e/test/scenarios/dashboard-filters/reproductions/8030-32444-reload-card-without-change.cy.spec.js]: nice catch!

**Pull request merged**